Sprint-1/implement/sum.test.js

```diff
@@ -10,27 +10,42 @@ const sum = require("./sum.js");
 
 // Acceptance Criteria:
 
-// Given an empty array
-// When passed to the sum function
-// Then it should return 0
-test.todo("given an empty array, returns 0")
-
-// Given an array with just one number
-// When passed to the sum function
-// Then it should return that number
-
-// Given an array containing negative numbers
-// When passed to the sum function
-// Then it should still return the correct total sum
-
-// Given an array with decimal/float numbers
-// When passed to the sum function
-// Then it should return the correct total sum
-
-// Given an array containing non-number values
-// When passed to the sum function
-// Then it should ignore the non-numerical values and return the sum of the numerical elements
-
-// Given an array with only non-number values
-// When passed to the sum function
-// Then it should return the least surprising value given how it behaves for all other inputs
+describe("sum", () => {
+  // Given an empty array
+  // When passed to the sum function
+  // Then it should return 0
+  test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+  });
+
+  // Given an array with just one number
+  // When passed to the sum function
+  // Then it should return that number
+  test("given an array with just one number, return that number", () => {
+    expect(sum([2])).toBe(2);
+  });
+  // Given an array containing negative numbers
+  // When passed to the sum function
+  // Then it should still return the correct total sum
+  test("given an array containing negative numbers, return the correct total sum", () => {
+    expect(sum([-1, -7, -10])).toBe(-18);
+  });
+  // Given an array with decimal/float numbers
+  // When passed to the sum function
+  // Then it should return the correct total sum
+  test("given an array with decimal/float numbers, return the correct total sum", () => {
+    expect(sum([1.0, 1.2, 1.5])).toBe(3.7);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So to test a calculated result that involves decimal numbers, we should use `.toBeCloseTo()` instead of `.toBe()`.
See: https://jestjs.io/docs/expect#tobeclosetonumber-numdigits